R/plot-res-vs-pred.R

```diff
@@ -78,3 +82,31 @@ plotResVsPred <- function(data,
   eval(parseUpdateAxes())
   return(plotObject)
 }
+
+
+#' @title plotResVsTime
+#' @description
+#' Producing residuals vs time plots
+#'
+#' @inheritParams plotObsVsPred
+#' @param dataMapping 
+#' A `ResVsTimeDataMapping` object mapping `x`, `y` and aesthetic groups to their variable names of `data`.
+#' @param plotConfiguration 
+#' An optional `ResVsTimeConfiguration` object defining labels, grid, background and watermark.
+#' @return A `ggplot` object
+#'
+#' @export
+#' @family molecule plots
+#' @examples 
+#' # Produce Obs vs Pred plot
+#' resVsTimeData <- data.frame(x = c(1, 2, 1, 2, 3), y = c(5, 0.2, 2, 3, 4))
+#' 
+#' plotResVsTime(data = resVsTimeData, dataMapping = ResVsTimeDataMapping$new(x = "x", y = "y"))
+#' 
+#' # Produce Res vs Time plot with linear regression
+#' plotResVsTime(
+#' data = resVsTimeData, 
+#' dataMapping = ResVsTimeDataMapping$new(x = "x", y = "y"),
+#' smoother = "lm"
+#' )
+plotResVsTime <- plotResVsPred
```

Comment:
I don't understand this.
Here `plotResidualsVsTime `function is just set equal to `plotResidualsVsPredicted `function?
So calling it would produce again residuals vs predicted plot and not residuals vs time plot???

Reply:
Before users had to use `plotResVsPred` and 
- define `x` = predicted in their dataMapping if the plot was actually residuals vs predicted.
- define `x` = time in their dataMapping if the plot was actually residuals vs time.

Having the exact same function for `plotResVsTime` does not change this workflow.
Users still need to map the `x` variable, which drives if the actual plot is residuals vs time or predicted.
For this reason, I did not think there is a need to "copy/paste" the function.
Using an alias seems a better solution to provide a function with a name consistent to what the plot display.

If you think this might be more confusing in the end, I can duplicate the code

Reply:
no, no code duplications pls :)
ok, so in fact we have just an XY-Scatter-Plot function with different "aliases" (plotResVsPred, plotResVsTime)?
Well, why not.

Reply:
Yes, they are both XY scatter plots + a line at y=0 + optional smoother/regression